src/azure-cli/azure/cli/command_modules/appconfig/_kv_helpers.py

```diff
@@ -391,6 +404,17 @@ def __read_kv_from_config_store(azconfig_client,
         count += 1
         if count >= top:
             return retrieved_kvs
+
+    # A request to list kvs of a non-existent snapshot returns an empty result.
+    # We first check if the snapshot exists before returning an empty result.
+    if snapshot and len(retrieved_kvs) == 0:
+        try:
+            _ = AppConfigSnapshotClient(azconfig_client).get_snapshot(name=snapshot)
+
+        except HttpResponseError as exception:
+            if exception.status_code == StatusCodes.NOT_FOUND:
+                raise ResourceNotFoundError("No snapshot with name '{}' was found.".format(snapshot))
```

Comment:
What do you think about moving this code inside the `if snapshot` check on line 349? It would keep all snapshot related logic in one place.

Reply:
Right, what I meant was we could check if configsetting_iterable has no items as soon as we call list_snapshot_kv. If there are no items, call get_snapshot to verify if snapshot exists or not.

Reply:
Got it! This has been updated. 
Since the "configsetting_iterable" is a paged iterator that makes calls to backend to fetch subsequent pages, we only confirm if it is empty by attempting to add to process the first value.
Kindly let me know your thoughts on this.

Reply:
You're right, this got too complicated and prone to errors in future. I think your previous approach was better.

Reply:
Got it, I will revert this change then.

Reply:
This has been updated.